sktime/forecasting/compose/_pipeline.py

```diff
@@ -1310,7 +1312,11 @@ def _fit(self, y, X, fh):
         # remember if X seen was None
         self.X_was_None_ = X is None
 
-        if self.behaviour == "update" and X is not None:
+        if (
+            self.behaviour == "update"
+            and X is not None
+            and not self.get_tag("ignores-exogeneous-X")
```

Comment:
I believe if the tag is `ignores-exogenous-X`, we already pass `X=None` to both `_fit` and `_predict`, so the third line is superfluous.

Reply:
Based on my changes in `_fit`, `self.forecaster_X_` do not exist if `forecaster_y` has `ignores-exogeneous-X` tag as `True`. So, if I don't add this check in `_update`, I thought it'll fail with `AttributeError`.

Now I see it's handled in `_check_X_y` in `BaseForecaster`, which I was not aware of. Will remove from both `_fit` and `_update` in that case, and the change in `_predict_*` (or in `_get_forecaster_X_prediction`) is also redundant in that case in my understanding. Let me know if I understood correctly.

Reply:
makes sense. We should e mindful that there are cases where `X` in `predict` and `update` can e none, ut in `fit` it is not (case of no future-known variales)